docs/source/write_first_app.rst

```diff
@@ -101,889 +97,485 @@ bringing up a new one.
+
+.. note:: In order to maintain security of any private data used in transactions, the application should connect to a
+          Gateway peer belonging to the same organization as the client identity. If the client identity's organization
+          does not host any peers then a Gateway peer hosted by an upstream organization through which the client's
+          organization is authorized to transact should be used.
+
+The TypeScript application creates a gRPC connection using the TLS certificate of the Certificate Authority for the
+Gateway peer's organization, so that the authenticity of the peer's credentials can be verified. Since the Docker
+container hosting the Gateway peer is reached at a ``localhost`` address, a gRPC option is specified to force this
+endpoint address to be interpreted as the hostname in the peer's TLS certificate, allowing the TLS connection to be
+successfully established.
+
+.. code-block:: TypeScript
+
+  const peerEndpoint = 'localhost:7051';
+
+  async function newGrpcConnection(): Promise<grpc.Client> {
+      const tlsRootCert = await fs.readFile(tlsCertPath);
+      const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+      return new grpc.Client(peerEndpoint, tlsCredentials, {
+          'grpc.ssl_target_name_override': 'peer0.org1.example.com',
+      });
+  }
 
-.. code:: bash
+Second, the application creates a Gateway connection
+----------------------------------------------------
 
-  // in a real application this would be done only when a new user was required to be added
-  // and would be part of an administrative flow
-  await registerAndEnrollUser(caClient, wallet, mspOrg1, org1UserId, 'org1.department1');
+The application then creates a Gateway connection, which the application uses to access any of the networks
+(analogous to channels) accessible to the Gateway peer, and smart contracts deployed to those networks. A Gateway
+connection requires three things:
 
-Similar to the admin enrollment, this function uses a CSR to register and enroll ``appUser`` and
-store its credentials alongside those of ``admin`` in the wallet. We now have
-identities for two separate users --- ``admin`` and ``appUser`` --- that can be
-used by our application.
+  1. gRPC connection to the Gateway peer.
+  2. Client identity used to transact with the network.
+  3. Signing implementation used to generate digital signatures for the client identity.
 
-Scrolling further down in your terminal output, you should see confirmation of the app user registration
-similar to this:
+In addition to the above, a hash implementation can also be specified. The default implementation will typically be
+sufficient.
 
-.. code:: bash
+The sample application uses an X.509 certificate for the Org1 user identity, and a signing implementation based on the
+corresponding private key. These credentials are created using the Org1 Certificate Authority when the test network
+is started.
```

Comment:
So needs something like "The following code snippet ... " is what, or does what.